### PR TITLE
[release-4.13] OCPBUGS-27649,OCPBUGS-27564,OCPBUGS-27654,OCPBUGS-27569: bump go-git/v5 to 5.11.0

### PR DESCRIPTION
**Pull request by anik120**
Fix for [CVE-2023-49568](https://bugzilla.redhat.com/show_bug.cgi?id=2258165) and [CVE-2023-49569](https://bugzilla.redhat.com/show_bug.cgi?id=2258143) for OLM and operator-registry.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -265,9 +265,6 @@ replace (
 	// latest tag resolves to a very old version. this is only used for spinning up local test registries
 	github.com/docker/distribution => github.com/docker/distribution v0.0.0-20191216044856-a8371794149d
 
-	// pin to ginkgo v1 version of gomega (v1.18+ relies on ginkgo v2)
-	github.com/onsi/gomega => github.com/onsi/gomega v1.17.0
-
```
Comment (anik120) [inline go.mod:270]: This was introduced by DanS [here](https://github.com/openshift/operator-framework-olm/commit/275408b28812e5b9d70bd72acbfffbb66e087b96), that was removed in [4.14](https://github.com/openshift/operator-framework-olm/blob/release-4.14/go.mod). Had to remove it from 4.13 manually to get dependencies to resolve correctly. This was not a part of @grokspawn's commit.

File: vendor/gomodules.xyz/jsonpatch/v2/jsonpatch.go [hunk 2]
```diff
@@ -24,21 +23,28 @@ func (j *Operation) Json() string {
...
-		v, err := json.Marshal(j.Value)
-		if err != nil {
-			return nil, err
-		}
-		b.WriteString(`,"value":`)
-		b.Write(v)
-	}
-	b.WriteString("}")
-	return b.Bytes(), nil
+	// Ensure for add and replace we emit `value: null`
```
Comment (anik120) [inline vendor/gomodules.xyz/jsonpatch/v2/jsonpatch.go:26]: There are changes to the vendor folder after pulling in @ankitathomas's commit, which is new for 4.13. There were no changes to the vendor folder after pulling in the commit until 4.14.

**Review by anik120 — COMMENTED** [review]
Things to `Note`:

**Review by ankitathomas — APPROVED** [review]
/lgtm

**Pull request merged**